### PR TITLE
Fix for custom cached disk creation

### LINKED ISSUE #57825: `filesystem_caches_path` in the configuration should be defined by default
1. Allow relative paths for the value of `filesystem_caches_path`. It should be relative to `path`.
2. Set it to `cache` by default in clickhouse-server's configuration to avoid a minor vulnerability with creating tables with dynamic disks.
3. Make sure it applies to dynamic disks, but statically configured paths are still allowed to point anywhere.
4. Make sure clickhouse-local still allows dynamic disks to have paths point anywhere.
5. Update the code of ClickBench, related to `clickhouse-web`.

### PR DESCRIPTION
**Pull request by kssenii**
### Changelog category (leave one):
- Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Make sure that for custom (created from SQL) disks ether `filesystem_caches_path` (a common directory prefix for all filesystem caches) or `custom_cached_disks_base_directory` (a common directory prefix for only filesystem caches created from custom disks) is specified in server config.  `custom_cached_disks_base_directory` has higher priority for custom disks over  `filesystem_caches_path`, which is used if the former one is absent.  Filesystem cache setting `path` must lie inside that directory, otherwise exception will be thrown preventing disk to be created. This will not affect disks created on an older version and server was upgraded - then the exception will not be thrown to allow the server to successfully start). `custom_cached_disks_base_directory` is added to default server config as `/var/lib/clickhouse/caches/`. Closes https://github.com/ClickHouse/ClickHouse/issues/57825.

### CODE REVIEW

**Comment by robot-ch-test-poll1** [comment]
*This is an automated comment for commit 8decf5039a3d62f9d98a64c88d86bb5f17d35820 with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/58869/8decf5039a3d62f9d98a64c88d86bb5f17d35820/ci_running.html) to open a full report in a separate page

<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/58869/8decf5039a3d62f9d98a64c88d86bb5f17d35820/ci_running.html">⏳ pending</a></td></tr>
<tbody>
</table>

File: programs/server/config.xml [hunk 1]
```diff
@@ -440,6 +440,9 @@
     <!-- Cache size in elements for compiled expressions.-->
     <compiled_expression_cache_elements_size>10000</compiled_expression_cache_elements_size>
 
+    <!-- Cache path for custom (created from AST) cached disks -->
```
Comment (vdimir) [inline programs/server/config.xml:443]:
```suggestion
    <!-- Cache path for custom (created from SQL) cached disks -->
```

File: src/Disks/ObjectStorages/Cached/registerDiskCache.cpp [hunk 2]
```diff
@@ -49,7 +51,40 @@ void registerDiskCache(DiskFactory & factory, bool /* global_skip_access_check *
...
-        if (config_fs_caches_dir.empty())
+        if (custom_disk)
+        {
+            static constexpr auto custom_cached_disks_base_dir_in_config = "custom_cached_disks_base_directory";
+            auto custom_cached_disk_path_prefix = context->getConfigRef().getString(custom_cached_disks_base_dir_in_config, config_fs_caches_dir);
+            if (custom_cached_disk_path_prefix.empty())
+            {
+                if (!attach)
+                {
+                    throw Exception(
```
Comment (vdimir) [inline src/Disks/ObjectStorages/Cached/registerDiskCache.cpp:62]: How it works if only `filesystem_caches_path` is set? We will check that `custom_cached_disk_path_prefix.empty()` and `!attach` and throw the exception

**Review by vdimir — APPROVED** [review]

File: src/Disks/ObjectStorages/Cached/registerDiskCache.cpp (hunk 2 above)
Comment (kssenii) [reply src/Disks/ObjectStorages/Cached/registerDiskCache.cpp:62]: See `auto custom_cached_disk_path_prefix = context->getConfigRef().getString(custom_cached_disks_base_dir_in_config, config_fs_caches_dir);` here `config_fs_caches_dir` is the default and it is equal to `filesystem_caches_path` from config.

**Comment by kssenii** [comment]
> Upgrade check (tsan) — Logical error thrown

https://github.com/ClickHouse/ClickHouse/issues/57893

> Stateless tests (release, analyzer) 

Not related to changes

**Pull request merged**